src/components/Table/hoc/withTableSettings/TableColumnSetup/TableColumnSetup.tsx

```diff
@@ -283,9 +303,19 @@ export const TableColumnSetup = (props: TableColumnSetupProps) => {
         className,
         defaultItems = propsItems,
         showResetButton: propsShowResetButton,
+        filterable,
+        filterPlaceholder,
+        filterEmptyMessage,
+        filterSettings = defaultFilterSettingsFn,
     } = props;
 
     const [open, setOpen] = React.useState(false);
+    const [sortingEnabled, setSortingEnabled] = React.useState(sortable);
+    const [prevSortingEnabled, setPrevSortingEnabled] = React.useState(sortable);
+    if (sortable !== prevSortingEnabled) {
+        setPrevSortingEnabled(sortable);
+        setSortingEnabled(sortable);
+    }
```

Comment:
Why changing state during render and not in effect?

Reply:
It should work faster than with useEffect. Component will be rerendered immediately after returning JSX before DOM updates. 
https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes